hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/recovery/ReconOMMetadataManager.java

```diff
@@ -47,4 +50,26 @@ public interface ReconOMMetadataManager extends OMMetadataManager {
    * @return true if OM Tables are initialized, otherwise false.
    */
   boolean isOmTablesInitialized();
+
+  /**
+   * Return all volumes in the file system.
+   * @return all the volumes from the OM DB.
+   */
+  List<OmVolumeArgs> listVolumes() throws IOException;
+
+  /**
+   * Check if volume exists in the OM table.
+   * @param volName volume name without any protocol prefix.
+   * @return true if volume exists, otherwise false.
+   * @throws IOException IOE
+   */
+  boolean volumeExists(String volName) throws IOException;
+
+  /**
+   * List all buckets under a volume.
+   * @param volumeName volume name without protocol prefix.
+   * @return buckets under volume or all buckets if volume is null.
+   */
+  List<OmBucketInfo> listBucketsUnderVolume(
```

Comment:
we should not load all bucket list in memory. Please use pagination and index concept. Need to keep the system bounded.

Reply:
Updated with limit (default 1000) to bound the number buckets/volumes returned.